### ISSUE TITLE
Reason of sudden RTH

### ISSUE THREAD

**Issue by IakovlevAA** [dji-sdk/Mobile-SDK-Android-V5#386]
Hi Dji,
I've encountered strange behaviour on Mini 3 Pro. When I'm using drone in simulator, in one moment KeyFlightMode changes to GO_HOME and it flies back to takeoff position, altough I didn't call RTH function. I've already disabled Intelligent RTH with KeyLowBatteryRTHEnabled set to false. but the problem still ocurs. FailSafeAction is FailsafeAction.HOVER. KeyIsFailSafe always false too
Steps to reproduce
1. Start simulator
2. Takeoff
3. Fly ~20m from takeoff position
4. Wait some time
5. RC starts beeping and drone return to home

**Comment by mitchrsm** [comment]
In my experience I believe it's the mini 3 pro overheating in simulator mode... , that too bad cause give very short time to test app functionality..

**Comment by IakovlevAA** [comment]
I'm using vent to cool drone, so I think it's not the issue. What I've noticed is, as low is battery charge, faster it returns to home, although intelligent RTH is disabled.

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/111119'>Zendesk ticket #111119</a>:<br><br>You can check the SDK for the received health information. The DJI MINI 3 PRO may experience overheating in a stationary state, triggering the overheating protection. You can use a small fan or other devices to help the aircraft dissipate heat for extended simulated testing.
https://developer.dji.com/api-reference-v5/android-api/Components/IDeviceHealthManager/IDeviceHealthManager.html#idevicehealthmanager_adddjidevicehealthinfochangelistener_inline
<br><p><em><small>°°°</small></em></p>

**Comment by IakovlevAA** [comment]
Yes, it indeed was overheating, fan was just not so effective... But how can I know if drone is overheated during flight? I 've just tried DeviceHealthManager and DeviceStatusManager, both told that there is no error. Also thought that BatteryKey.KeyBatteryTemperature can tell about overheating, but didn't help

**Comment by dji-lyt** [comment]
![image](https://github.com/dji-sdk/Mobile-SDK-Android-V5/assets/104296295/fc1e3295-749c-425a-906a-4c7c8689e02a)
The DeviceHealthManager should provide information when the aircraft is overheating.
During flight, the aircraft's cooling system is much more effective than when it is stationary, so the likelihood of overheating is lower.

**Comment by IakovlevAA** [comment]
On Mini 3 Pro can't reproduce. Only got one warning about low speed sd card, however drone started returning home due to overheating
![image](https://github.com/dji-sdk/Mobile-SDK-Android-V5/assets/43800616/2df0bc9c-bc70-431c-9f12-c4c3ad33aff1)

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/111119'>Zendesk ticket #111119</a>:<br><br>Is the MSDK version and drone firmware version you are using the latest? Were there any relevant prompts before the aircraft returned?

If the SDK version and firmware are not up to date, could you try updating to the latest version?
<br><p><em><small>°°°</small></em></p>

**Comment by IakovlevAA** [comment]
Right now I'm using 5.9.0, but already tried with 5.9.1, no changes. There were no prompts before,while and after aircraft returned

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/111119'>Zendesk ticket #111119</a>:<br><br>I seem to have reproduced a same issue. Please wait for a while.
<br><p><em><small>°°°</small></em></p>

**Comment by antonymarion** [comment]
Yes I stumbled too over the same behaviour, and it took me time to understand that mini 3 Pro was overheating and there was no fan (that flying with it replaces the fan effect :D )

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/111119'>Zendesk ticket #111119</a>:<br><br>The Mini 3 series will include an overheat RTH prompt in version 5.11. We plan to release version 5.11 in November 2024, but the exact release date is currently uncertain.
<br><p><em><small>°°°</small></em></p>
